driver-async/src/main/com/mongodb/async/client/ChangeStreamIterable.java

```diff
@@ -29,7 +30,7 @@
  * @mongodb.server.release 3.6
  * @since 3.6
  */
-public interface ChangeStreamIterable<TResult> extends MongoIterable<TResult> {
+public interface ChangeStreamIterable<TResult> extends MongoIterable<ChangeStreamOutput<TResult>> {
```

Comment:
Given the restrictions of ChangeStreamOutput and its interaction with the aggregate pipeline, which could change the shape of the document, this class could use an explanatory comment about that, and point to the rawResults method as an escape hatch.  And rawResults would benefit from a comment about the need to preserve the exact shape of the resumeToken.